spec/factories/competitions.rb

```diff
@@ -355,7 +355,10 @@
     after(:create) do |competition|
       competition.delegates.each do |delegate|
         unless delegate.region_id.nil? # There can be cases where the competition delegate is actually not a delegate (temporary delegate)
-          FactoryBot.create(:senior_delegate, region_id: delegate.region_id)
+          if UserGroup.find(delegate.region_id).lead_user.nil? # Allowing to manually create senior delegate for the delegate if needed.
+            senior_delegate_role = FactoryBot.create(:senior_delegate_role)
+            delegate.update!(region_id: senior_delegate_role.group.id)
```

Comment:
When is the value that's passed to this `update` ever different from the original `region_id`?

You're only entering this block if an "original" `region_id` is present (i.e. not nil). Only in that case are you trying to access the `UserGroup`, and only if that user group can be found you're creating a senior delegate for that exact user group.

The Senior Delegate you created will by definition always be the Senior Delegate for the original `region_id`, so I don't understand why this update is necessary.

Reply:
Basically, if a delegate doesn't have a senior delegate, I'm creating a senior delegate here. But when I create a senior delegate using `FactoryBot.create(:senior_delegate_role)`, it's internally creating a group as well. And then I'm changing the real delegate's group to the new group. I've now changed it the reverse - like creating the senior delegate with the same group of the delegate. I don't remember which was the test affected, but if all tests pass then it's good to go. Otherwise I'll check on it.